source/agora/test/Base.d

```diff
@@ -1589,8 +1590,19 @@ private mixin template TestNodeMixin ()
     /// Return an enrollment manager backed by an in-memory SQLite db
     protected override EnrollmentManager makeEnrollmentManager ()
     {
-        return new EnrollmentManager(this.stateDB, this.cacheDB,
-            this.config.validator.key_pair, this.params);
+        import agora.consensus.PreImage;
+
+        if (this.config.validator.cycle_seed != Hash.init)
+        {
+            auto cycle = PreImageCycle(this.config.validator.cycle_seed,
+                this.config.validator.cycle_seed_height,
+                this.params.ValidatorCycle);
+            return new EnrollmentManager(this.stateDB, this.cacheDB,
+                this.config.validator.key_pair, this.params, cycle);
+        }
+        else
+            return new EnrollmentManager(this.stateDB, this.cacheDB,
+                this.config.validator.key_pair, this.params);
```

Comment:
This probably belongs to the base node class, doesn't it ?

Reply:
Moved it to `FullNode.d`.